modulus/distributed/utils.py

```diff
@@ -130,90 +130,91 @@ def gather_loss(loss: float, dst_rank: int = 0, mean: bool = True):
 
-#     # global transposition
-#     req = dist.all_to_all(x_recv, x_send, group=group, async_op=async_op)
+    # global transposition
+    req = dist.all_to_all(x_recv, x_send, group=group, async_op=async_op)
 
-#     return x_recv, req
+    return x_recv, req
 
 
-# def _reduce(input_, use_fp32=True, group=None):
-#     """All-reduce the input tensor across model parallel group."""
+def _reduce(input_, use_fp32=True, group=None):
+    """All-reduce the input tensor across model parallel group."""
 
-#     # Bypass the function if we are using only 1 GPU.
-#     if dist.get_world_size(group=group) == 1:
-#         return input_
+    # Bypass the function if we are using only 1 GPU.
+    if dist.get_world_size(group=group) == 1:
+        return input_
 
-#     # All-reduce.
-#     if use_fp32:
-#         dtype = input_.dtype
-#         inputf_ = input_.float()
-#         dist.all_reduce(inputf_, group=group)
-#         input_ = inputf_.to(dtype)
-#     else:
-#         dist.all_reduce(input_, group=group)
+    # All-reduce.
+    if use_fp32:
+        dtype = input_.dtype
+        inputf_ = input_.float()
+        dist.all_reduce(inputf_, group=group)
+        input_ = inputf_.to(dtype)
+    else:
+        dist.all_reduce(input_, group=group)
 
-#     return input_
+    return input_
 
 
-# def _split(input_, dim_, group=None):
-#     """Split the tensor along its last dimension and keep the corresponding slice."""
-#     # get input format
-#     input_format = get_memory_format(input_)
+def _split(input_, dim_, group=None):
+    """Split the tensor along its last dimension and keep the corresponding slice."""
+    # get input format
+    input_format = get_memory_format(input_)
 
-#     # Bypass the function if we are using only 1 GPU.
-#     comm_size = dist.get_world_size(group=group)
-#     if comm_size == 1:
-#         return input_
+    # Bypass the function if we are using only 1 GPU.
+    comm_size = dist.get_world_size(group=group)
+    if comm_size == 1:
+        return input_
 
-#     # Split along last dimension.
-#     input_list = split_tensor_along_dim(input_, dim_, comm_size)
+    # Split along last dimension.
+    input_list = split_tensor_along_dim(input_, dim_, comm_size)
 
-#     # Note: torch.split does not create contiguous tensors by default.
-#     rank = dist.get_rank(group=group)
-#     output = input_list[rank].contiguous(memory_format=input_format)
+    # Note: torch.split does not create contiguous tensors by default.
+    rank = dist.get_rank(group=group)
+    output = input_list[rank].contiguous(memory_format=input_format)
 
-#     return output
+    return output
 
 
-# def _gather(input_, dim_, group=None):
-#     """Gather tensors and concatinate along the last dimension."""
-#     # get input format
-#     input_format = get_memory_format(input_)
+def _gather(input_, dim_, group=None):
+    """Gather tensors and concatinate along the last dimension."""
```

Comment:
- While left over from previous versions, I think we should fix the typos of `concatinate` now in several places :) 
- `_gather` here is over the specified dimension, right, and not only the last dimension?
- I'm unsure about naming these utilities and their arguments following the `_` convention. I wouldn't oppose just a collection of "public" wrappers for gather/scatter etc. including their Autograd Wrappers. 
- With the comment above in mind, this should could be something like `all_gather_wrapper` instead of `_gather`

Reply:
Fixed the docstring issues. I agree about renaming the utilities, but wondering if we can do this in a follow-up PR. I would like to keep this PR fairly minimal since it is just reintroducing utilities. Would be good to keep actual changes separate from this PR.

Reply:
Created an issue to track this: https://github.com/NVIDIA/modulus/issues/112

Reply:
Thanks 👍🏽